src/devices/sonoff.ts

```diff
@@ -1174,6 +1186,210 @@ const sonoffExtend = {
+        };
+    },
+    setDimmingLightRate: (): ModernExtend => {
+        const exposes = e
+            .enum("dimming_light_rate", ea.ALL, ["1x", "2x", "3x", "4x", "5x"])
+            .withDescription("Speed of brightness change via external switch.");
+        const clusterName = "customClusterEwelink";
+        const attributeName = "dimmingLightRate";
+        const toZigbee: Tz.Converter[] = [
+            {
+                key: ["dimming_light_rate"],
+                convertSet: async (entity, key, value, meta) => {
+                    let rate = 0;
+                    if (value === "1x") {
+                        rate = 1;
+                    } else if (value === "2x") {
+                        rate = 2;
+                    } else if (value === "3x") {
+                        rate = 3;
+                    } else if (value === "4x") {
+                        rate = 4;
+                    } else if (value === "5x") {
+                        rate = 5;
+                    }
+                    logger.info(`from zigbee set dimming light rate: ${rate}`, NS);
+                    await entity.write<typeof clusterName, SonoffEwelink>(clusterName, {[attributeName]: rate}, defaultResponseOptions);
+                    return {state: {[key]: value}};
+                },
+                convertGet: async (entity, key, meta) => {
+                    await entity.read<typeof clusterName, SonoffEwelink>(clusterName, [attributeName], defaultResponseOptions);
+                },
+            },
+        ];
+        const fromZigbee = [
+            {
+                cluster: clusterName,
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    if (msg.data.dimmingLightRate !== undefined) {
+                        const rate = msg.data.dimmingLightRate;
+                        logger.info(`form zigbee dimming light rate ${rate}`, NS);
+
+                        let rateStatus = "1x";
+                        if (rate === 1) {
+                            rateStatus = "1x";
+                        } else if (rate === 2) {
+                            rateStatus = "2x";
+                        } else if (rate === 3) {
+                            rateStatus = "3x";
+                        } else if (rate === 4) {
+                            rateStatus = "4x";
+                        } else if (rate === 5) {
+                            rateStatus = "5x";
+                        }
+                        return {dimming_light_rate: rateStatus};
+                    }
+                },
+            } satisfies Fz.Converter<typeof clusterName, SonoffEwelink, ["attributeReport", "readResponse"]>,
+        ];
+        return {
+            exposes: [exposes],
+            fromZigbee,
+            toZigbee,
+            isModernExtend: true,
+        };
+    },
+    minBrightnessThreshold: (): ModernExtend => {
```

Comment:
Here `m.numeric` can be used with custom `fzConvert`, example: https://github.com/Koenkk/zigbee-herdsman-converters/blob/d3358036b4ab8e73d3a8876784846684ec7bc936/src/devices/sonoff.ts#L2316